contracts/dao/USDTLpListaDistributor.sol

```diff
@@ -0,0 +1,470 @@
+
+    emit USDTStaked(lpToken, usdtAmount, actualLpAmount);
+  }
+
+  /**
+   * @dev Unstake LP token from farming contract and withdraw USDT and lisUSD from PancakeStableSwapTwoPool
+   * @param lpAmount amount of LP token
+   * @param minLisUSDAmount minimum amount of lisUSD to withdraw
+   * @param minUSDTAmount minimum amount of USDT to withdraw
+   */
+  function withdraw(uint256 lpAmount, uint256 minLisUSDAmount, uint256 minUSDTAmount) external {
+    // 1. Validate lisUSD and USDT amount
+    (uint256 expectLisUSDAmount, uint256 expectUSDTAmount) = getCoinsAmount(lpAmount);
+    require(minLisUSDAmount <= expectLisUSDAmount, "Invalid lisUSD amount");
+    require(minUSDTAmount <= expectUSDTAmount, "Invalid USDT amount");
+
+    // 2. Update user's LP balance and LISTA reward, and distributor's LP total supply
+    _withdraw(msg.sender, lpAmount);
+
+    // 3. Unstake LP token from farming contract if not in emergency mode
+    if (!emergencyMode) _unstakeLp(msg.sender, lpAmount);
+
+    // 4. Remove liquidity from PancakeStableSwapTwoPool
+    uint256 lisUSDAmountActual = lisUSD.balanceOf(address(this));
+    uint256 usdtAmountActual = usdt.balanceOf(address(this));
+
+    IStableSwap(stableSwapPool).remove_liquidity(lpAmount, [minLisUSDAmount, minUSDTAmount]);
+
+    lisUSDAmountActual = lisUSD.balanceOf(address(this)) - lisUSDAmountActual;
+    usdtAmountActual = usdt.balanceOf(address(this)) - usdtAmountActual;
+    require(lisUSDAmountActual >= minLisUSDAmount, "Invalid lisUSD amount received");
+    require(usdtAmountActual >= minUSDTAmount, "Invalid USDT amount received");
+
+    // 5. Transfer lisUSD and USDT to user
+    lisUSD.safeTransfer(msg.sender, lisUSDAmountActual);
+    usdt.safeTransfer(msg.sender, usdtAmountActual);
+
+    emit LpUnstaked(lpToken, usdtAmountActual, lisUSDAmountActual, lpAmount);
+  }
+
+  /**
+   * @dev Harvest LP staking reward (CAKE) from farming contract
+   * @return claimed CAKE amount
+   */
+  function harvest() external whenNotPaused notInEmergencyMode returns (uint256) {
+    address distributor = address(this);
+
+    if (noHarvest()) return 0;
+
+    // Claim CAKE rewards from V2Wrapper
+    uint256 beforeBalance = IERC20(cake).balanceOf(distributor);
+    lastHarvestTime = block.timestamp;
+    v2wrapper.deposit(0, false);
+    uint256 claimed = IERC20(cake).balanceOf(distributor) - beforeBalance;
+
+    // Send CAKE to StakingVault
+    if (claimed > 0) {
+      IERC20(cake).safeApprove(stakeVault, claimed);
+      IStakingVault(stakeVault).sendRewards(distributor, claimed);
+      emit Harvest(address(usdt), distributor, claimed);
+    }
+
+    return claimed;
+  }
+
+  /**
+   * @dev claim staked LP reward (CAKE)
+   * @return reward amount
+   */
+  function claimStakeReward() external whenNotPaused returns (uint256) {
+    address _account = msg.sender;
+    uint256 amount = _claimStakingReward(_account);
+    IStakingVault(stakeVault).transferAllocatedTokens(_account, amount);
```

Comment:
Probably, Do this only if amount  is above 0

Reply:
thanks. done in 76fe49bc3623c3c682e3c5af0f602f5fca6320f1